core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java

```diff
@@ -235,6 +235,7 @@ public void writeEmptyIterator() throws Exception {
     final Option<MapStatus> mapStatus = writer.stop(true);
     assertTrue(mapStatus.isDefined());
     assertTrue(mergedOutputFile.exists());
+    assertEquals(0, spillFilesCreated.size());
```

Comment:
BTW, since we are touching `ShuffleExternalSorter.java`, it seems that we need to add a test case to `ShuffleExternalSorterSuite`, too.